sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,268 @@
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+
+// Get a reference to the blob address, then upload the file to the blob.
+CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(filename);
+await cloudBlockBlob.UploadFromFileAsync(localFilePath);
+```
+
+v12
+```c
+// Assumes containerClient already contains a reference to the container.
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+
+// Get a reference to a blob
+BlobClient blobClient = containerClient.GetBlobClient(filename);
+
+// Open the file and upload its data
+using FileStream uploadFileStream = File.OpenRead(localFilePath);
+await blobClient.UploadAsync(uploadFileStream, true);
+uploadFileStream.Close();
+```
+
+Summary: In v11, you would get the `CloudBlockBlob` reference by calling the `GetBlockBlobReference` method on the container name. Calling `UploadFromFileAsync` would then create the blob if it doesn't exist or overwrite the existing one. In v12, you will get the reference to the `BlobClient` object  by calling the `GetBlobClient` method on the container and then upload the blob using the `UploadAsync` method.
+
+
+### Downloading Blobs from a Container
+
+Legacy (v11)
+```c
+// Assumes you have already created a reference to the blob via blobClient
+// downloadFilePath should be the path to the intended file to download the blob to
+await cloudBlockBlob.DownloadToFileAsync(downloadFilePath, FileMode.Create);
+```
+
+v12
+```c
+// Assumes you have already created a reference to the blob via blobClient
+// downloadFilePath should be the path to the intended file to download the blob to
+BlobDownloadInfo download = await blobClient.DownloadAsync();
+
+using (FileStream downloadFileStream = File.OpenWrite(downloadFilePath))
+{
+    await download.Content.CopyToAsync(downloadFileStream);
+    downloadFileStream.Close();
+}
+```
+
+### Listing Blobs in a Container
+
+Legacy (v11)
+```c
+// List the blobs in the container.
+// Assumes a reference to the container via `cloudBlobContainer`
+BlobContinuationToken blobContinuationToken = null;
+do
+{
+    var results = await cloudBlobContainer.ListBlobsSegmentedAsync(null, blobContinuationToken);
+    // Get the value of the continuation token returned by the listing call.
+    blobContinuationToken = results.ContinuationToken;
+    foreach (IListBlobItem item in results.Results)
+    {
+        Console.WriteLine(item.Uri);
+    }
+} while (blobContinuationToken != null); // Loop while the continuation token is not null.
+```
+
+v12
+```c
+// Get a reference to the container
+BlobContainerClient container = new BlobContainerClient(connectionString, containerName);
+
+// List all blobs in the container
+await foreach (BlobItem blobItem in containerClient.GetBlobsAsync())
+{
+    Console.WriteLine("\t" + blobItem.Name);
+}
+```
```

Comment:
Do we have an example for .NET Framework applications that can't use `await foreach`?

Reply:
Do you mean that don't use it? Sorry I didn't quite understand the ask

Reply:
I think the ask is to provide a sync snippet and async snippet. But we already have samples for that. I think customers can look to our samples if they want to see how to do use async or sync versions of our APIs. Up to you if you want to provide both in the migration guide, but we can always link to our samples as well.